core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -292,6 +286,18 @@ private[spark] class CoarseMesosSchedulerBackend(
   }
 
 
+  def isOfferValidForScheduling(meetsConstraints: Boolean,
+                                slaveId: String, mem: Double,
+                                cpus: Int, sc: SparkContext): Boolean = {
+    taskIdToSlaveId.size < executorLimit &&
```

Comment:
Let's leave a TODO here that we want to abstract out each condition to have a specific error message logged why a offer doesn't meet the requirement.